docs/building-operators: update `make` usage in quickstarts and tutorials, general alignment (#4665)
operator-framework/operator-sdk: website/content/en/docs/building-operators/ansible/quickstart.md

- Go through the [installation guide][install-guide].
- User authorized with `cluster-admin` permissions.
- An accessible image registry for various operator images (ex. [hub.docker.com](https://hub.docker.com/signup),

> [quay.io](https://quay.io/)) and be logged in in your command line environment.

Reviewer: If we add that here then instead of example.com we would put quay.io in all.
However, should we add this as a pre-requirement? Could not the person use the docker.io instead? 
Because of this I still think that the best approach is to use `<some-registry>/` and add here the info that is required access to a registry such as quay.io for example.

> Make sure to define `IMG` when you call `make`:

Reviewer: Would not this one has the goal to close the issue: https://github.com/operator-framework/operator-sdk/issues/4415?

  ```sh

>   make docker-build docker-push IMG="example.com/memcached-operator:v0.0.1"

Reviewer: ```suggestion
  make docker-build docker-push IMG="<some-registry>/memcached-operator:v0.0.1"
```

Author: This value literally defaults to `example.com/memcached-operator` in this example, see https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/Makefile#L32.

Reviewer: Hi @estroz, 

See that the person needs to push the image to there registry. Otherwise, it does not work. So, based on the previous questions/issues raised about that the  `example.com` does not make clear it. PS.: Also, why someone that is following the quick start would like the testdata example?

IMO the test data example require to be `example.com` because we cannot flag/set any specific registry. So, after the person read the tutorial and check `<some-registry>/memcached-operator:v0.0.1` and then see `example.com/memcached-operator:v0.0.1"` in the sample then, that information is very clear too.

Author: There's an [entire section](https://github.com/operator-framework/operator-sdk/pull/4665/files#diff-904511a83c884e3a04cbd1183f9b91e6f730dc5cdeacdd00ba654e66545b8146R133) about this in the tutorial. Perhaps a note at the top of each quickstart, but nothing more. There's only so much hand-holding we can do here.

  ```

> 

Reviewer: ```suggestion
NOTE: To allow the cluster to pull the image the repository needs to be set as public or you must configure an image pull secret.
```

Author: This shouldn't be in the quickstart.

Reviewer: We need to add that in the quick start. Otherwise, the users will not understand that the image needs are pushed and pullable. it is a common scenario/issue faced by them following the previous instructions of the operator-sdk-sample that was solved by providing this info.

Author: If they are using a private image registry, they definitely know about pull secrets.

Reviewer: For me does not matter the place in the doc. However, I think we need to clarify that is required to be able to push and pull the image.

  operator-sdk olm install
  ```

> 1. Bundle your operator, then build and push the bundle image (defaults to `example.com/memcached-operator-bundle:v0.0.1`):

Reviewer: ```suggestion
1. Bundle your operator, then build and push the bundle image (defaults to `<some-registry>/memcached-operator-bundle:v0.0.1`):
```

  ```sh

>   make bundle IMG="example.com/memcached-operator:v0.0.1"

Reviewer: ```suggestion
  make bundle IMG="<some-registry>/memcached-operator:v0.0.1"
```

  make bundle-build bundle-push
  ```

1. Run your bundle:

  ```sh

>   operator-sdk run bundle example.com/memcached-operator-bundle:v0.0.1

Reviewer: ```suggestion
  operator-sdk run bundle <some-registry>/memcached-operator-bundle:v0.0.1
```

1. Deploy your operator:

  ```sh

>   make deploy IMG="example.com/memcached-operator:v0.0.1"

Reviewer: ```suggestion
  make deploy IMG="<some-registry>/memcached-operator:v0.0.1"
```
